### ISSUE TITLE
temporal_extent in load_collection includes the second element

### ISSUE THREAD

**Issue by clausmichele** [Open-EO/openeo-odc#4]
The openEO API https://processes.openeo.org/#load_collection specifies that the second element should be excluded.

**Comment by danielFlemstrom** [comment]
a temporal_extent from 2021-06-01 00:00 to 2021-06-01 23:59 gives no result since of a whole day is subtracted in map_processes_ocs.py::exclusive_date, a few seconds should do the trick (assuming the problem is that OpenEO is "up to" and the cube is "up to and including"). 
When you have several images a day and want one image, subtracting a whole day makes things difficult for us. 
<img width="991" alt="image" src="https://user-images.githubusercontent.com/12410550/168019115-f81b6660-b5e6-430a-99c8-e2ddc7488701.png">
Would it be possible to use second instead of day here or would that cause other problems that I cannot see ?

**Comment by clausmichele** [comment]
Good point, @m-mohr could you please tell me if subtracting 1 second behavior would still be API compliant?

**Comment by m-mohr** [comment]
I think you nay need to subtract depending on the granularity of the input.

If you receive 2020-01-01T00:00:00Z you'd need to subtract a second so that you get 2019-12-31T23:59:59Z.
If you receive 2020-01-01 you'd need to subtract a day so that you get 2019-12-31 (at 0:00:00).

**Comment by danielFlemstrom** [comment]
Should it not be 2019-12-13 (23:59) ?

**Comment by m-mohr** [comment]
That's a good question. Having a closer look, the process description doesn't define this. Actually, you could probably also just remove 1 s always and be compliant as it's undocumented right now. I think actually my first comment is likely not the common understanding, but the more common understand when people read this is what @danielFlemstrom says. So maybe indeed just remove a second always.

**Comment by danielFlemstrom** [comment]
Just a clarification of our problem. This is an excerpt of the actual time sequence of a product we have:
2020-10-01 08:20:09 
2020-10-01 09:21:39 
2020-10-01 09:24:39 
2020-10-01 10:01:08 
2020-10-01 10:04:08 
2020-10-01 11:02:38 
2020-10-02 08:55:28 
2020-10-02 09:34:57 
2020-10-02 09:37:57 
2020-10-02 10:34:05 
So if you want to look at ONE of these pictures, say 2020-10-01 09:21:39 , it is quite tricky to get it right.

**Comment by clausmichele** [comment]
If you modify this line https://github.com/Open-EO/openeo-odc/blob/095d9c287d9a98ee77d421c31cb63cb741d42d02/src/openeo_odc/map_processes_odc.py#L60 with
`            return np.datetime_as_string(np.datetime64(date) - np.timedelta64(1, 's'), timezone='UTC') # Substracts one second`

and you provide in the openEO process graph a temporal extent of `["2020-10-01T09:21:39Z","2020-10-01T09:22:39Z"]`  it should work fine!

**Comment by danielFlemstrom** [comment]
Any chance of getting that fix into the repo ?
